src/lib/util.ts

```diff
@@ -169,6 +169,18 @@ export function shouldSkipConfirm(options?: MethodOptions): boolean {
   return !!options.skipConfirm;
 }
 
+export function shouldRelay(
```

Comment:
Use the same logic to decide if caller wants to relay for all methods

Reply:
@joewagner Random TS question as I'm still learning the syntax/intricacies.

Since `shouldRelay` can take `WriteOptions` or `SetControllerOptions` as well, does it make sense to use a "discriminated union"? Idk if that makes things more complicated and/or isn't really necessary in TS world.

E.g., something like an `Options` type is the param in `shouldRelay`, instead of  `MethodOptions`. Or maybe this is what `MethodOptions` could be? Along these lines:
```
type Options = WriteOptions | SetControllerOptions | MethodOptions
```

Reply:
@dtbuchholz Good call! I can update this unless you want to get in on this PR?  If so, feel free to commit to this branch.  Otherwise I'll update this tonight

Reply:
Sure, I can take a stab

Reply:
So, I ended up adding a few different discriminated unions. I'm not sure if this adds unnecessary types for the sake of readability.

For example, if the union is `type RelayOptions = WriteOptions | SetControllerOptions`, instead, this could be done in the function signature -- e.g., `options: RelayOptions` could instead be `options: WriteOptions | SetControllerOption`. No strong preference here, not sure what's a better practice.

Reply:
FYI, `npm i` from both `main` and this PR's branch throws some error, complaining about mismatch with `ethers` and `siwe`. It works if I use the `--force` flag, though, so I figured I'd share for visibility purposes:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: siwe@1.1.6
npm ERR! Found: ethers@5.6.9
npm ERR! node_modules/ethers
npm ERR!   ethers@"^5.5.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer ethers@"5.5.1" from siwe@1.1.6
npm ERR! node_modules/siwe
npm ERR!   siwe@"^1.1.6" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: ethers@5.5.1
npm ERR! node_modules/ethers
npm ERR!   peer ethers@"5.5.1" from siwe@1.1.6
npm ERR!   node_modules/siwe
npm ERR!     siwe@"^1.1.6" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /Users/db/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/db/.npm/_logs/2022-08-12T00_07_04_250Z-debug-0.log
```